h2quic/server.go

```diff
@@ -20,10 +20,8 @@ import (
 )
 
 type streamCreator interface {
-	AcceptStream() (utils.Stream, error)
+	quic.Session
```

Comment:
I think the normal pattern is to have only one function in these types of interfaces, and to type assert onto it just before the method call. It doesn't matter, just fyi.